### DISCUSSION TITLE
Jinja2Temmplates does not accept PosixPath on Python 3.7 of Raspberry OS

### DISCUSSION THREAD

**Discussion by dumtux** [fastapi/fastapi#8675] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates

# on Raspberry OS, this is a PosixPath object
STATIC_DIR = pathlib.Path(__file__).parent / "static"

app = FastAPI()

# StaticFiles accept PosixPath object
app.mount("/static", StaticFiles(directory=STATIC_DIR), name="static")

# error, Jinja2Templates does not accept PosixPath object
templates = Jinja2Templates(directory=STATIC_DIR / "template")

# # for fast fix, need to cast PosixPath object to str
# templates = Jinja2Templates(directory=str(STATIC_DIR / "template"))
```

### Description

I am using Jinja2 for building simple frontend of my FastAPI app.

```python
STATIC_DIR = pathlib.Path(__file__).parent / "static"
templates = Jinja2Templates(directory=STATIC_DIR / "template")
```
`Jinja2Templates` works fine with `pathlib.Path` on Ubuntu, Python 3.9.
But when I tried the same code on my Raspberry Pi, it gave the error.

```
...
  File "/usr/local/lib/python3.7/dist-packages/starlette/templating.py", line 53, in __init__
    self.env = self.get_env(directory)
  File "/usr/local/lib/python3.7/dist-packages/starlette/templating.py", line 61, in get_env
    loader = jinja2.FileSystemLoader(directory)
  File "/usr/lib/python3/dist-packages/jinja2/loaders.py", line 163, in __init__
    self.searchpath = list(searchpath)
TypeError: 'PosixPath' object is not iterable
```

I could avoid this error by casting the path object to `str`.

### Operating System

Linux

### Operating System Details

**Raspberry OS**,, Apr 30, 2021 update

### FastAPI Version

0.68.1

### Python Version

3.7.3

**Comment by Kludex** [comment, marked as answer]
This issue should be on Starlette, jfyk.
